common/persistence/sql/sqlplugin/interfaces.go

```diff
@@ -54,8 +54,7 @@ type VersionedBlob struct {
 type (
 	// Plugin defines the interface for any SQL database that needs to implement
 	Plugin interface {
-		CreateDB(dbKind DbKind, cfg *config.SQL, r resolver.ServiceResolver, l log.Logger, mh metrics.Handler) (DB, error)
-		CreateAdminDB(dbKind DbKind, cfg *config.SQL, r resolver.ServiceResolver, l log.Logger, mh metrics.Handler) (AdminDB, error)
+		CreateDB(dbKind DbKind, cfg *config.SQL, r resolver.ServiceResolver, l log.Logger, mh metrics.Handler) (any, error)
```

Comment:
Unfortunately, Go does not support type parameters here.

Reply:
would it help to put a comment on this function saying that the `any` is the created database? (since that can no longer be inferred from the function signature)

I mean, maybe it could be inferred simply from the fact that this is a create method, but idk just an idea!

Reply:
I can add that. I considered using an empty interface, but `DB` is already taken. It's really quite odd that you can just cast your way to the DB interface you need here.

Reply:
Added a `GenericDB` interface with the methods that both share instead, now.